.travis.yml

```diff
@@ -20,8 +20,15 @@ addons:
 
 before_script:
 - |
-  pip install 'travis-cargo<0.2' --user &&
-  export PATH=$HOME/.local/bin:/usr/local/bin:$PATH
+  if [ $TRAVIS_OS_NAME = 'osx' ]; then
+    brew install python3
+    virtualenv env -p python3
+    source env/bin/activate
+    pip install 'travis-cargo<0.2'
+  else
+    pip install 'travis-cargo<0.2' --user
+  fi
+- export PATH="$(python -m site --user-base)/bin:$PATH"
```

Comment:
Is the idea here to get the `travis-cargo` script onto `PATH`? In the virtualenv case (macOS here), `travis-cargo` is not in `$(python -m site --user-base)/bin`. It's in `$VIRTUAL_ENV/bin`. However, you don't need to add anything to the path in this case because `virtualenv` modifies the path when you activate it. Thus, you should be able to move this export line into your `else` block.

# `virtualenv` case
```sh
# let's peek at PATH before activating the virtualenv
$ echo $PATH
/Users/davidalber/.cargo/bin:<other stuff>

$ virtualenv env -p python3
# (output omitted)

$ . env/bin/activate

# let's take another look at PATH
$ echo $PATH
/Users/davidalber/env/bin:/Users/davidalber/.cargo/bin:<other stuff>

# contrast with VIRTUAL_ENV
$ echo $VIRTUAL_ENV
/Users/davidalber/env

$ ls $VIRTUAL_ENV/bin
activate                easy_install            pip3.6                  python3.6
activate.csh            easy_install-3.6        python                  wheel
activate.fish           pip                     python-config
activate_this.py        pip3                    python3

# ^^ no `travis-cargo` script

$ pip install 'travis-cargo<0.2'
# (output omitted)

$ ls $VIRTUAL_ENV/bin
activate                easy_install            pip3.6                  python3.6
activate.csh            easy_install-3.6        python                  travis-cargo
activate.fish           pip                     python-config           wheel
activate_this.py        pip3                    python3

# ^^ there it is!
```

# Non-`virtualenv` case
_I did this in an Ubuntu 16.04 Docker container._
```sh
$ python -m site --user-base
/home/myuser/.local

$ ls `python -m site --user-base`/bin
ls: cannot access '/home/myuser/.local/bin': No such file or directory

$ pip install 'travis-cargo<0.2' --user
# (output omitted)

$ ls `python -m site --user-base`/bin
travis-cargo
```

Reply:
You are indeed correct. I'm learning on my feet here, and am a Windows user just brute forcing things. This is a clear improvement on what I have, thanks!